scipy/stats/_stats.pyx

```diff
@@ -513,6 +513,99 @@ cdef double _geninvgauss_pdf(double x, void *user_data) nogil except *:
     return math.exp(_geninvgauss_logpdf_kernel(x, p, b))
 
 
+cdef double _phi(double z) nogil:
+    """evaluates the normal PDF. Used in `studentized range`"""
+    cdef double inv_sqrt_2pi = 0.3989422804014327
+    return inv_sqrt_2pi * math.exp(-0.5 * z * z)
+
+
+cdef double _Phi(double z) nogil:
+    """evaluates the normal CDF. Used in `studentized range`"""
+    return cs.ndtr(z)
+
+cdef double _genstudentized_range_cdf(int n, double[2] x, void *user_data) nogil:
```

Comment:
cdef double _genstudentized_range_cdf(**int n,** double[2] x, void *user_data) nogil:


@DominicChm out of curiosity what is the function of `int n`?

Reply:
I think I asked that in another comment. I think it's the number of element in `x`.